input/drivers_joypad/sdl_joypad.c

```diff
@@ -55,10 +56,10 @@ static const char *sdl_joypad_name(unsigned pad)
 
 #ifdef HAVE_SDL2
    if (sdl_pads[pad].controller)
-      return SDL_GameControllerNameForIndex(pad);
-   return SDL_JoystickNameForIndex(pad);
+      return SDL_GameControllerNameForIndex(sdl_pads[pad].sdl_id);
```

Comment:
Not really needed, but we could likely use [SDL_GameControllerName](https://wiki.libsdl.org/SDL2/SDL_GameControllerName) here.
``` suggestion
      return SDL_GameControllerName(sdl_pads[pad].controller);
```